facebookresearch/habitat-sim
# [WIP] Test emscripten build for WebGL deployment

## Motivation and Context

Preliminary testing of habitat-sim compiling through emscripten for WebGL deployment.  Currently, just a "pile of hacks" to get viewer utility binary working only with pre-loaded data package containing test scenes.

Before landing will need to:
- [ ] Figure out proper handling of dynamic asset loading (example dynamic data package loading [here](https://github.com/kripken/BananaBread/blob/master/cube2/js/game-setup.js))
- [ ] Remove hack ignoring `texSize` setting (see [here](https://github.com/facebookresearch/habitat-sim/blob/decouple-util-deps/src/esp/gfx/GenericShader.cpp#L166-L169))
- [x] Also compile with headless contexts so that things other than viewer utility work
- [ ] Lots of other stuff ...

## How Has This Been Tested

1. Download and install emscripten SDK from https://emscripten.org/
1. Make sure to source emscripten env variables before building (i.e. `source ~/code/emsdk/emsdk_env.sh` or similar)
1. Run `./build_js.sh`
1. Follow instructions at end of build to launch server and open viewer in browser tab

## Types of changes



## Checklist

- [X] I have completed my CLA (see **CONTRIBUTING**)
